src/vos/vos_obj.c

```diff
@@ -745,6 +748,26 @@ key_iter_fetch(struct vos_obj_iter *oiter, vos_iter_entry_t *ent,
 	if (rc != 0)
 		return rc;
 
+	if (check_existence && oiter->it_iter.it_filter_cb != NULL &&
+	    (flags & VOS_ITER_PROBE_AGAIN) == 0) {
```

Comment:
I'm not sure why we want to skip pre_check on 'PROBE_AGAIN'?

I suppose all pre_checks will be used to accelerate filtering? so calling it again here to accelerate filtering and increase credits looks even better to me.

Reply:
I was thinking of potentially moving the ilog pre-aggregate check to the filter.  I guess it depends on how expensive the filtering call could get.  My thinking was that we already ran the checks on the object or key so why do it again?

Reply:
Ok, I'm fine with either approach.